Filter out _removed revision from push replication

pasin opened the pull request:
- Filter out _removed revision in CBL_Pusher's processInbox when preparing revisions to send to the remote server.

- No changes in CBL_Pusher.unpushedRevisions properties. Therefore unpushedRevisions could contained _removed revisions. The same for getting _changes feed from the CBLRouter.

- Added a unit test

- Added cbl_replicator_removed_rev remote database config to Unit-Tests/Server/cbl_unit_tests.json

#671

snej commented on Source/CBL_Pusher.m:
```diff
@@ -279,6 +279,12 @@ - (void) processInbox: (CBL_RevisionList*)changes {
                             continue;
                         }
+                        
+                        if ([loadedRev[@"_removed"] boolValue]) {
```
This will throw an exception if the `_removed` property contains anything but a NSNumber. Probably best to compare it to `@YES`.

snej commented on Unit-Tests/Replication_Tests.m:
```diff
@@ -969,4 +970,96 @@ - (void) test16_Restart {
+
+    NSError* error;
+
+    // Create a new document with grant = true:
```
This seems too complex for a unit test. It would be fine to just manually create a revision with a `"_removed":true` property and then verify that it doesn't get pushed.

pasin commented:
@snej I have updated the code according to your review comment. I also rebased with the current master. 

To check _removed value, I use `$castIf(NSNumber, loadedRev[@"_removed"]).boolValue` the same way as the other places.
